programs/vote/src/vote_state/mod.rs

```diff
@@ -1072,10 +1274,11 @@ pub fn process_vote_state_update<S: std::hash::BuildHasher>(
     vote_account: &KeyedAccount,
     slot_hashes: &[SlotHash],
     clock: &Clock,
-    vote_state_update: VoteStateUpdate,
+    mut vote_state_update: VoteStateUpdate,
     signers: &HashSet<Pubkey, S>,
 ) -> Result<(), InstructionError> {
     let mut vote_state = verify_and_get_vote_state(vote_account, clock, signers)?;
+    vote_state.check_update_vote_state_slots_are_valid(&mut vote_state_update, slot_hashes)?;
```

Comment:
This also needs a feature gate

Reply:
Same as https://github.com/solana-labs/solana/pull/22358#discussion_r781723344